src/lambda_handler.py

```diff
@@ -34,54 +164,253 @@ def task_default(event):
     return result
 
 
-def task_cowsay(event) -> dict[str, Union[Optional[str], bool]]:
-    """Generate an output message using the provided information."""
-    result: dict[str, Union[Optional[str], bool]] = {"message": None, "success": True}
+class FileConfig(TypedDict):
+    """Define the type structure of the dictionary in the file_configs variable."""
 
-    character: str = event.get("character", "tux")
-    if character not in cowsay.characters.CHARS.keys():
-        error_msg = 'Character "%s" is not valid.'
-        logging.error(error_msg, character)
-        failed_task(result, error_msg % character)
+    app_regex: re.Pattern[str]
+    description: str
+    filename: str
+    # I'd prefer to define ip_set as Set[Union[IPv4Address, IPv6Address]],
+    # but ipaddress.collapse_addresses() uses the type variable "_N" while
+    # ip_network returns Union[IPv4Network, IPv6network] which causes
+    # the mypy pre-commit hook to throw this error:
+    #   Value of type variable "_N" of "collapse_addresses" cannot be
+    #   "Union[IPv4Network, IPv6Network]"
+    # My solution to this problem is to simply define ip_set as Set[Any].
+    # For a similar issue and discussion, see
+    # https://github.com/python/typeshed/issues/2080
+    ip_set: Set[Any]
+    static_ips: List[str]
+
+
+def validate_event_data(
+    event: Dict[str, Any]
+) -> Tuple[Dict[str, Any], bool, List[str]]:
+    """Validate the event data and return a tuple containing the validated event, a boolean result (True if valid, False if invalid), and a list of error message strings."""
+    result = True
+    errors = []
+
+    # Account ID checks
+    if "account_ids" not in event:
+        errors.append('Missing required key "account_ids" in event.')
+    elif event["account_ids"] == []:
+        errors.append('"account_ids" must be a non-empty list.')
     else:
-        contents: str = event.get("contents", "Hello from AWS Lambda!")
-        logger.info(
-            'Creating output using "%s" with contents "%s"', character, contents
-        )
-        result["message"] = cowsay.get_output_string(character, contents)
+        account_ids: List[str] = event["account_ids"]
+        try:
+            # Ensure account_ids is a list of strings
+            if type(account_ids) is not list:
```

Comment:
So this actually came up when I was using `pylint` to clean up another project. `isinstance()` is preferred vs `type()` because it is both faster and considers object inheritance. So this would validate not just a `list` but also if it is a subclass of `list` whereas `type()` only checks if it is a `list`. We should only use `type()` if we want to check if something is a specific class and not a subclass. Not particularly impactful here but a good best practice to get into.

```suggestion
            if not isinstance(account_ids, list):
```